### DISCUSSION TITLE
Slint application crashes after MacOS codesign?

### DISCUSSION THREAD

**Discussion by oswald2** [slint-ui/slint#4323] [Q&A] [answered]
Hello,

I have a Rust slint application (using also rfd) that is compiled under Linux, Windows and MacOS, which is working perfectly fine. For a distribution on Mac you need to codesign and notarize it. 

I use cargo bundle to create a bundle and it still works. When I do the codesign, afterwards the application starts and seems to work, but on the first call to a file dialog it crashes with a PoisonError (which according to Rust doc is when a thread fails while a mutex is held). 

I normally don't work on Mac and I am quite new to this, so general question: how do you create working Mac apps using slint and rfd (I did just a checkout from git and compile and use VS Code for small edits)? Do you use XCode (how to setup)? Has anybody had the same issue? If so, how to survive the codesign?

**Comment by tronical** [comment, marked as answer]
My best guess here is that now your app is missing certain entitlements. Could it be that it is sandboxed by now? The activity monitor can show if you enable the sandbox column in the view menu.

**Comment by oswald2** [comment]
Sorry for the late response. Yes, you were correct it was sandboxed. Removed that and it worked. Thanks a lot!
